Backends/Wayland/port.lisp

```diff
@@ -0,0 +1,332 @@
+  ;; Is this where I should set the egl-window width and swap buffers?
+  ;; Should this be handled by the medium? or te mirror?
+  (alx:when-let ((mirror (sheet-mirror sheet)))
+    (format *debug-io* "handle-event window-cfg-event: top level reconfigure? Should I resize all the children?~%")
+    ;; QQQQ Should I use port-set-mirror-geometry or is there some other way
+    ;; to trigger it?
+    ;; (break)
+    (unless (eql +everywhere+ (window-event-region event))
+     (with-bounding-rectangle* (x1 y1 x2 y2 :width w :height h)
+         (window-event-region event)
+       (with-bounding-rectangle* (ox1 ox2 oy1 oy2)
+           ;; QQQQ why is this -5 1 -5 1 still
+           (sheet-mirror-geometry sheet)
+         (when (some #'/=
+                     (list x1 y1 x2 y2)
+                     (list ox1 oy1 ox2 oy2))
+           (format *debug-io* "resize surface geometry new:~s current:~s~%"
+                   (list x1 y1 x2 y2)
+                   (list ox1 oy1 ox2 oy2))
+           ;; TODO: determine if we need CLX's ROUND-COORDINATE
+           (xdg:xdg-surface-set-window-geometry (slot-value (port sheet) 'surface) x1 y1 w h)
+
+           (wl-egl:wl-egl-window-resize
+            (wayland-egl-mirror-window mirror) w h x1 y1)
+           ;; Is this overkill?
+           (port-force-output (port sheet))))))))
+
+(defmethod graft-width ((graft wayland-graft) &key (units :device))
+  (let ((screen (wayland-port-device (port graft))))
+    (ecase units
+      (:device (screen-width screen))
+      (:inches (/ (screen-physical-width screen) 25.4s0))
+      (:millimeters (screen-physical-width screen))
+      (:screen-sized 1))))
+
+(defmethod graft-height ((graft wayland-graft) &key (units :device))
+  (let ((screen (wayland-port-device (port graft))))
+    (ecase units
+      (:device (screen-height screen))
+      (:inches (/ (screen-physical-height screen) 25.4s0))
+      (:millimeters (screen-physical-height screen))
+      (:screen-sized 1))))
+
+(defmethod make-graft ((port wayland-port) &key (orientation :default)
+                                             (units :device))
+  (let* ((screen (wayland-port-device port))
+         (root   (wayland-port-window port))
+         (mirror (make-instance 'wayland-egl-mirror :window root))
+         (width  (screen-width screen))
+         (height (screen-height screen))
+         (region (make-bounding-rectangle 0 0 width height)))
+    (make-instance 'wayland-graft :port port
+                                  :region region
+                                  :mirror mirror
+                                  :orientation orientation
+                                  :units units)))
+
+(defclass wayland-frame-manager (standard-frame-manager)
+  ((mirroring :initarg :mirroring
+              :initform nil
+              :reader mirroring)
+   (class-gensym :initarg :class-gensym
+                 :initform (gensym "WAYLAND-")
+                 :reader class-gensym)))
```

Comment:
I wouldn't cargo-cult the frame manager, using standard-frame-manager directly should be fine (then only top level sheet mixin gets a mirror by default)